server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -263,6 +278,224 @@ TransportResponse handleExtensionRequest(ExtensionRequest extensionRequest) {
+
+        final CountDownLatch inProgressLatch = new CountDownLatch(1);
+
+        // initialize map of entries for this extension to return
+        final Map<DiscoveryNode, Map<Class<?>, Map<String, ExtensionReader>>> extensionRegistry = new HashMap<>();
+
+        final TransportResponseHandler<NamedWriteableRegistryResponse> namedWriteableRegistryResponseHandler = new TransportResponseHandler<
+            NamedWriteableRegistryResponse>() {
+
+            @Override
+            public NamedWriteableRegistryResponse read(StreamInput in) throws IOException {
+                return new NamedWriteableRegistryResponse(in);
+            }
+
+            @Override
+            public void handleResponse(NamedWriteableRegistryResponse response) {
+
+                logger.info("response {}", response);
+
+                logger.info("EXTENSION [" + extensionNode.getName() + "] returned " + response.getRegistry().size() + " entries");
+                if (response.getRegistry().isEmpty() == false) {
+
+                    // Extension has sent over entries to register, initialize inner category map
+                    Map<Class<?>, Map<String, ExtensionReader>> categoryMap = new HashMap<>();
+
+                    // reader map associated with this current category
+                    Map<String, ExtensionReader> readers = null;
+                    Class currentCategory = null;
+
+                    // Extract response entries and process fully qualified class name into category class instance
+                    for (Map.Entry<String, String> entry : response.getRegistry().entrySet()) {
+
+                        try {
+                            String name = entry.getKey();
+                            Class<?> categoryClass = Class.forName(entry.getValue());
+
+                            if (currentCategory != categoryClass) {
+                                // after first pass, readers and current category are set
+                                if (currentCategory != null) {
+                                    categoryMap.put(currentCategory, readers);
+                                }
+                                readers = new HashMap<>();
+                                currentCategory = categoryClass;
+                            }
+
+                            // add name and callback method reference to inner reader map
+                            // Since appending the new reader entry returns the previous value associated with key, or null if there was no
+                            // mapping for key, validate that name has not yet been associated with a callback method to it's associated
+                            // extension
+                            ExtensionReader newReader = (en, cc, context) -> parseNamedWriteable(en, cc, context);
+                            ExtensionReader oldReader = readers.put(name, newReader);
+                            if (oldReader != null) {
```

Comment:
Better way to check
```suggestion
                            if (oldReader != null && oldReader.getClass()) {
```

Reply:
(I think you need another != null there?)